k2/csrc/cuda/log_test.cu

```diff
@@ -0,0 +1,63 @@
+// k2/csrc/cuda/log.h
+
+// Copyright (c)  2020  Fangjun Kuang (csukuangfj@gmail.com)
+
+// See ../../../LICENSE for clarification regarding multiple authors
+
+#include "gtest/gtest.h"
+#include "k2/csrc/cuda/log.h"
+
+namespace k2 {
+
+TEST(Log, Cpu) {
+  K2_LOG(DEBUG) << "Debug message";
+  K2_LOG(INFO) << "Info message";
+  K2_LOG(WARNING) << "Warning message";
+  K2_LOG(ERROR) << "Error message";
+
+  int a = 10;
+  int b = 20;
+  int c = 30;
+  int d = a;
+
+  K2_CHECK(a == d) << "failed";
+
+  K2_CHECK_EQ(a, d) << "failed";
+  K2_CHECK_NE(a, b) << "failed";
+
+  K2_CHECK_LE(a, b) << "failed";
+  K2_CHECK_LT(a, b) << "failed";
+
+  K2_CHECK_GE(c, b) << "failed";
+  K2_CHECK_GT(c, b) << "failed";
+}
+
+__global__ void DummyKernel(int *b, int a) {
+  K2_LOG(INFO) << "In kernel";
+  K2_CHECK_LT(*b, a) << K2_KERNEL_DEBUG_STR;
```

Comment:
maybe it's better if `CHECK` can print out `K2_KERNEL_DEBUG_STR` automatically when called in kernel

Reply:
@csukuangfj Bear in mind that a lot of the time we write kernel code in lambda expressions that could get evaluated on the kernel or on the CPU. 
Also I'd rather not have to do explicitly `<< K2_KERNEL_DEBUG_STR` every time, if possible.

Reply:
When more threads are here, like this `<<<2, 2>>>`, and to compare with printf, add a printf line below. The code used is:
```cpp
__device__ void DummyKernel(int *b, int a) {
  K2_LOG(INFO) << "In kernel";
  K2_CHECK_EQ(*b, a) << K2_KERNEL_DEBUG_STR << " asdfsad " << a << " = a";
  printf("[%d, %d, %d]: a=%d, b=%d\n", threadIdx.x, threadIdx.y, threadIdx.z, a, *b);
  *b += 1;
  K2_CHECK_EQ(*b, a) << K2_KERNEL_DEBUG_STR;
}

__global__ void EntryKernel(int *b, int a) {
  DummyKernel(b, a);
}
.
.
.
  EntryKernel<<<2, 2>>>(b, a);
```
I got the results below:
![image](https://user-images.githubusercontent.com/705582/92222601-d78ae100-eed1-11ea-8688-730bc9e97d8b.png)

That indicate overload the "<<" would make msg mess up with theads as each "<<" call printf one-time.

Reply:
> maybe it's better if CHECK can print out K2_KERNEL_DEBUG_STR automatically when called in kernel

Done.

There is also a DCHECK.